test/e2e/common_suite_test.go

```diff
@@ -48,14 +61,25 @@ func (tc *testCase) withSecret(secret *v1.Secret) *testCase {
 	tc.secret = secret
 	return tc
 }
-
+func (tc *testCase) withJob(job *batchv1.Job) *testCase {
+	tc.job = job
+	return tc
+}
+func (tc *testCase) withPod(pod *v1.Pod) *testCase {
+	tc.pod = pod
+	return tc
+}
 func (tc *testCase) withTestCommands(testCommands []testCommand) *testCase {
 	tc.testCommands = testCommands
 	return tc
 }
+func (tc *testCase) withExpectedPodLogString(expectedPodLogString string) *testCase {
+	tc.expectedPodLogString = expectedPodLogString
+	return tc
+}
 
 func (tc *testCase) run() {
-	podFeature := features.New(fmt.Sprintf("%s Pod", tc.pod.Name)).
+	podFeature := features.New(fmt.Sprintf("%s test", tc.testName)).
```

Comment:
This is an optional change but we could rename `podFeature` and `Create pod` message on the line below this one to reflect that the test cases can create objects other than single pods. E.g., `testCaseFeature` and `"Create testworkload"`.